lib/Crypto/generaterandomdatarequest.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2018 Jolla Ltd.
+ * Contact: Chris Adams <chris.adams@jollamobile.com>
+ * All rights reserved.
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#ifndef LIBSAILFISHCRYPTO_GENERATERANDOMDATAREQUEST_H
+#define LIBSAILFISHCRYPTO_GENERATERANDOMDATAREQUEST_H
+
+#include "Crypto/cryptoglobal.h"
+#include "Crypto/request.h"
+#include "Crypto/key.h"
+
+#include <QtCore/QObject>
+#include <QtCore/QScopedPointer>
+#include <QtCore/QString>
+#include <QtCore/QByteArray>
+
+namespace Sailfish {
+
+namespace Crypto {
+
+class CryptoManager;
+
+class GenerateRandomDataRequestPrivate;
```

Comment:
Why not to put it inside the `GenerateRandomDataRequest` class itself?

Reply:
Private implementation class allows us to make changes later which would otherwise be binary incompatible.  I will do similar pimpl-isation of the rest of the API in the near future.

Reply:
I know what is PIMPL. I ask why we could not put the line `class GenerateRandomDataRequestPrivate` inside the `class GenerateRandomDataRequest {` so it would look like:

```c++
class GenerateRandomDataRequest
{
//....
private:
    class GenerateRandomDataRequestPrivate;
};
```